pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessorTest.java

```diff
@@ -1903,19 +1908,76 @@ public void testForwardIndexDisabledOnExistingColumnDictEncoded()
     ColumnMetadata columnMetadata = segmentMetadata.getColumnMetadataFor(EXISTING_STRING_COL_DICT);
     assertNotNull(columnMetadata);
 
-    SegmentDirectory segmentDirectory = SegmentDirectoryLoaderRegistry.getDefaultSegmentDirectoryLoader()
-        .load(_indexDir.toURI(),
-            new SegmentDirectoryLoaderContext.Builder().setSegmentDirectoryConfigs(_configuration).build());
-    SegmentPreProcessor v3Processor =
-        new SegmentPreProcessor(segmentDirectory, _indexLoadingConfig, _newColumnsSchemaWithForwardIndexDisabled);
-    expectThrows(RuntimeException.class, v3Processor::process);
+    createAndValidateIndex(ColumnIndexType.FORWARD_INDEX, EXISTING_STRING_COL_DICT, 9, 4,
+        _newColumnsSchemaWithForwardIndexDisabled, false, true, false, 26, true, 0, null, true, DataType.STRING,
+        100000);
 
     constructV1Segment(Collections.emptyList(), Collections.emptyList(), Collections.emptyList());
-    segmentDirectory = SegmentDirectoryLoaderRegistry.getDefaultSegmentDirectoryLoader().load(_indexDir.toURI(),
-        new SegmentDirectoryLoaderContext.Builder().setSegmentDirectoryConfigs(_configuration).build());
-    SegmentPreProcessor v1Processor =
-        new SegmentPreProcessor(segmentDirectory, _indexLoadingConfig, _newColumnsSchemaWithForwardIndexDisabled);
-    expectThrows(RuntimeException.class, v1Processor::process);
+
+    createAndValidateIndex(ColumnIndexType.FORWARD_INDEX, EXISTING_STRING_COL_DICT, 9, 4,
```

Comment:
Can we please add another test here where the existing column contains RangeIndex.
After reload forward index should be disabled and range index should be reconstructed?

Reply:
Not sure I follow. Why should rangeIndex need to be reconstructed after disabling forward index ?

Upon deleting the forward index, regardless of whether range index is dict based or raw value based, it should continue to work during queries.

Reply:
For disabling forward index on a raw column, I gave a comment earlier about whether to update tableConfig. Since we will implicitly create dictionary, in that range index will have to be reconstructed. 

But first we may want to discuss what to do for disabling forward index on raw column

Reply:
I guess I wasn't clear enough 😓 . I meant that we should add a test for the following case:

T1: Column  C1 has raw forwardIndex, range index (raw)
T2: ForwardIndex is disabled
T3: Segment is reloaded.
Now, during reload, the following operations should happen:
1. Temporary dict based forward index is created.
2. Old raw range index is deleted.
3. Dictionary and inverted index are created 
4. New dict based range index is created.

Reply:
done